### PR TITLE
[17] token interceptor

### PR DESCRIPTION
**Pull request by steve1rm**
# Pull Request template

###  What does this PR solve?

- Adds Token Interceptor

### CODE REVIEW

File: data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package me.androidbox.data.remote.interceptor
+
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class TokenInterceptor(private val preferenceRepository: PreferenceRepository) : Interceptor {
+
+    private companion object {
+        const val AUTHORIZATION_HEADER = "Authorization"
```
Comment (philipplackner) [inline data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt]: This string will never change and this constant also doesn't add extra readability, so in that case I personally prefer to leave it

**Review by philipplackner — APPROVED** [review]

File: data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                .newBuilder()
+                .addHeader(AUTHORIZATION_HEADER, "Bearer ${loginUser.token}")
+                .build()
+
+            chain.proceed(request)
+        }
+        else {
+            chain.proceed(chain.request())
+        }
+    }
```
Comment (MKiperszmid) [inline data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt]: This is my personal opinion:

I usually prefer to not have != if later I'm going to add an else

So right now:

if (loginUser!= null) {} else {}

I prefer having a == null and then an else. 

So I would do:

```

return if (loginUser == null) {
    chain.proceed(chain.request())
} else {
    val request = chain.request()
                .newBuilder()
                .addHeader(AUTHORIZATION_HEADER, "Bearer ${loginUser.token}")
                .build()
    chain.proceed(request)
}
```

**Review by MKiperszmid — APPROVED** [review]

File: data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt (hunk 1 above)
Comment (steve1rm) [reply data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt]: Updated to use the string directly

File: data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt (hunk 2 above)
Comment (steve1rm) [reply data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt]: Updated the to check for == null in the if statement

**Pull request merged**